testDataNodeRestartWithBusyMasterDuringSnapshot fails in CI

The `DedicatedClusterSnapshotRestoreIT#testDataNodeRestartWithBusyMasterDuringSnapshot` test fails in a reproducible manner. This failure was seen in CI https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+intake/522/console and reproduced locally for me. 

Reproduce:

```
./gradlew :server:integTest \
  -Dtests.seed=D10290B601FA94B1 \
  -Dtests.class=org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT \
  -Dtests.method="testDataNodeRestartWithBusyMasterDuringSnapshot" \
  -Dtests.security.manager=true \
  -Dtests.locale=hi \
  -Dtests.timezone=America/Barbados \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Error:

```
java.lang.AssertionError: IN_PROGRESS
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:847)
	at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.testDataNodeRestartWithBusyMasterDuringSnapshot(DedicatedClusterSnapshotRestoreIT.java:1191)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
	Suppressed: java.lang.AssertionError: IN_PROGRESS
		at org.junit.Assert.fail(Assert.java:88)
		at org.junit.Assert.assertTrue(Assert.java:41)
		at org.elasticsearch.snapshots.DedicatedClusterSnapshotRestoreIT.lambda$testDataNodeRestartWithBusyMasterDuringSnapshot$8(DedicatedClusterSnapshotRestoreIT.java:1196)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:835)
		... 38 more
```

--- a/server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java
+++ b/server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java
@@ -1161,7 +1161,6 @@ public class DedicatedClusterSnapshotRestoreIT extends AbstractSnapshotIntegTest
         assertThat(anotherStats.getTotalSize(), is(snapshot1FileSize));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/36283")
     public void testDataNodeRestartWithBusyMasterDuringSnapshot() throws Exception {
         logger.info("-->  starting a master node and two data nodes");
         internalCluster().startMasterOnlyNode();
@@ -1203,7 +1202,7 @@ public class DedicatedClusterSnapshotRestoreIT extends AbstractSnapshotIntegTest
             assertEquals(1, snapshotsStatusResponse.getSnapshots().size());
             SnapshotInfo snapshotInfo = snapshotsStatusResponse.getSnapshots().get(0);
             assertTrue(snapshotInfo.state().toString(), snapshotInfo.state().completed());
-        }, 30, TimeUnit.SECONDS);
+        }, 60L, TimeUnit.SECONDS);
     }
 
     private long calculateTotalFilesSize(List<Path> files) {

